syft/pkg/cataloger/golang/package.go

```diff
@@ -6,18 +6,25 @@ import (
 	"strings"
 
 	"github.com/anchore/packageurl-go"
+	"github.com/anchore/syft/internal/log"
 	"github.com/anchore/syft/syft/pkg"
 	"github.com/anchore/syft/syft/source"
 )
 
-func newGoBinaryPackage(dep *debug.Module, mainModule, goVersion, architecture string, buildSettings map[string]string, locations ...source.Location) pkg.Package {
+func (c *goBinaryCataloger) newGoBinaryPackage(resolver source.FileResolver, dep *debug.Module, mainModule, goVersion, architecture string, buildSettings map[string]string, locations ...source.Location) pkg.Package {
 	if dep.Replace != nil {
 		dep = dep.Replace
 	}
 
+	licenses, err := c.licenses.getLicenses(resolver, dep.Path, dep.Version)
```

Comment:
Do we want licenses to be nil here or, if there is an err, have it be an initialized empty array?

Reply:
Good call -- this always returns a non-nil slice now